### PR TITLE
C++ implementation of issue #1665 and issue #1674

### PR DESCRIPTION
**Pull request by mike-lischke**
Second attempt at that patch. The first patch exists but seems to be dangling in space. Could not create a PR from that.

### CODE REVIEW

File: runtime/Cpp/runtime/src/Parser.h [hunk 1]
```diff
@@ -376,6 +377,30 @@ namespace antlr4 {
...
+
+    /** How to create an error node, given a token, associated with a parent.
+       *  Typically, the error node to create is not a function of the parent
+       *  but this method must still set the parent pointer of the terminal node
+       *  returned. I would prefer having {@link ParserRuleContext#addAnyChild(ParseTree)}
+       *  set the parent pointer, but the parent pointer is implementation dependent
+       *  and currently there is no setParent() in {@link ErrorNode} (and can't
+       *  add method in Java 1.7 without breaking backward compatibility).
+       *
+       * @since 4.6.1
```
Comment (parrt) [inline runtime/Cpp/runtime/src/Parser.h]: Could you change the 4.6.1 to 4.7?

File: runtime/Cpp/runtime/src/tree/TerminalNode.h [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace tree {
      *  TerminalNodes anyway so I'm adding as it improves internal
      *  code quality.
      *
-     *  @since 4.6.1
+     *  @since 4.7
```
Comment (parrt) [inline runtime/Cpp/runtime/src/tree/TerminalNode.h]: you might do a search for the other 4.6.1s :)

Comment (mike-lischke) [inline runtime/Cpp/runtime/src/tree/TerminalNode.h]: done

**Comment by mike-lischke** [comment]
@parrt ready for merge?

**Pull request merged**